tests/featureWriters/kernFeatureWriter_test.py

```diff
@@ -667,44 +681,69 @@ def test_kern_LTR_and_RTL(self, FontClass):
                 sub reh-ar by reh-ar.fina;
                 language URD;
             } fina;
+
+            feature isol {
+                script arab;
+                sub alef-ar by alef-ar.isol;
+            } isol;
             """
         )
 
         ufo = makeUFO(FontClass, glyphs, groups, kerning, features)
 
         newFeatures = self.writeFeatures(ufo, ignoreMarks=False)
 
-        assert str(newFeatures) == dedent(
-            """\
-            @kern1.A = [A Aacute];
-            @kern1.reh = [reh-ar zain-ar reh-ar.fina];
-            @kern2.alef = [alef-ar alef-ar.isol];
+        assert dedent(str(newFeatures)) == dedent(
+            """
+            lookup kern_Arab {
+                pos four-ar seven-ar -30;
+                pos reh-ar.fina lam-ar.init <-80 0 -80 0>;
+                pos [reh-ar reh-ar.fina zain-ar] [alef-ar alef-ar.isol] <-100 0 -100 0>;
+            } kern_Arab;
 
-            lookup kern_dflt {
-                pos seven four -25;
-            } kern_dflt;
+            lookup kern_Latn {
+                enum pos [A Aacute] V -40;
+            } kern_Latn;
 
-            lookup kern_ltr {
-                enum pos @kern1.A V -40;
-            } kern_ltr;
+            lookup kern_Thaa {
+                pos four-ar seven-ar -30;
+            } kern_Thaa;
 
-            lookup kern_rtl {
+            lookup kern_Yezi {
                 pos four-ar seven-ar -30;
-                pos reh-ar.fina lam-ar.init <-80 0 -80 0>;
-                pos @kern1.reh @kern2.alef <-100 0 -100 0>;
-            } kern_rtl;
+            } kern_Yezi;
+
+            lookup kern_Common {
+                pos seven four -25;
+            } kern_Common;
 
             feature kern {
-                lookup kern_dflt;
-                script latn;
+                script DFLT;
                 language dflt;
-                lookup kern_ltr;
-                language TRK;
+                lookup kern_Common;
+
                 script arab;
                 language dflt;
-                lookup kern_rtl;
-                language URD;
+                lookup kern_Common;
+                lookup kern_Arab;
+
+                script latn;
+                language dflt;
+                lookup kern_Common;
+                lookup kern_Latn;
+
+                script thaa;
+                language dflt;
+                lookup kern_Common;
+                lookup kern_Thaa;
             } kern;
+
+            feature dist {
+                script yezi;
+                language dflt;
+                lookup kern_Common;
+                lookup kern_Yezi;
+            } dist;
```

Comment:
Where does the test pick up the Thaa and Yezi scripts? They don't seem to appear in the original test, and that makes a new dist feature as well?

Reply:
As part of https://github.com/googlefonts/ufo2ft/pull/679/commits/427aa80aa05b89d4cb79c43a2a79fef7c07c1f61, we now look at _all_ scripts a codepoint is used in and just generate lookups for all of them, rather than guessing what the font supports or the designer intended.

Reply:
You can see if the glyphs is used in multiple scripts and only consider the ones declared in the feature file (like I’m doing in #678). If the feature file doesn’t declare any scripts at all, or the character is used only in one script then it is a fair game and you can continue doing what you do now.

Reply:
@khaledhosny I'm looking at your PR now. I had this other idea pop into my head, what about checking the glyphset for entries with a single associated script (e.g. U+0780 THAANA LETTER HAA) and multiple associated scripts (e.g. U+061F ARABIC QUESTION MARK) and then cutting down the multiple scripts to the found single scripts? Would you know of a case where this goes wrong?

Something like, for NotoSansThaana-Regular.ufo:

```txt
In [10]: single = set()
    ...: for glyph in u:
    ...:     for cp in glyph.unicodes or []:
    ...:         scripts = unicodedata.script_extension(chr(cp))
    ...:         if len(scripts) == 1:
    ...:             single.update(scripts)
    ...:

In [11]: single
Out[11]: {'Arab', 'Thaa', 'Zinh', 'Zyyy'}
```

(some glyphs like U+061F ARABIC QUESTION MARK contribute the single Arab script). Then, all multi-script sets would be cut down to the intersection of `single`.

Reply:
That is one way to do it, and I think it should work. One downside is that the font designer has less control (they can’t see I want to support` Thaa` as well without including Thaana glyphs, but it probably does not matter in practice.